In [1]:
import pandas as pd



## Carga de Datos
Cargamos los datasets necesarios: `df_final_experiment_clients.txt`

In [2]:
df = pd.read_csv('df_final_experiment_clients.txt')


print(f"Experiment Clients: {df.shape}")


Experiment Clients: (70609, 2)


##  Renombrado y Tipos de Datos
Renombramos `Variation` a `experiment_group` para mayor claridad y verificamos los tipos de datos.

In [3]:
# Renombrar columna
df.rename(columns={'Variation': 'experiment_group'}, inplace=True)

# Verificación de tipos de datos
print("Tipos de datos actuales:")
print(df.dtypes)



Tipos de datos actuales:
client_id            int64
experiment_group    object
dtype: object


## Análisis Exploratorio

In [4]:
# Muestra de datos
display(df.head())

client_id experiment_group
0    9988021             Test
1    8320017             Test
2    4033851          Control
3    1982004             Test
4    9294070          Control

In [5]:
# Análisis de la columna 'experiment_group' (Categorías)
print("Valores únicos en la columna experiment_group:")
print(df['experiment_group'].value_counts(dropna=False))

Valores únicos en la columna experiment_group:
experiment_group
Test       26968
Control    23532
NaN        20109
Name: count, dtype: int64


In [6]:
# Conteo de Nulos en experiment_group
nulls = df['experiment_group'].isnull().sum()
print(f"Nulos en experiment_group: {nulls} ({df['experiment_group'].isnull().mean() * 100:.2f}%) ")

Nulos en experiment_group: 20109 (28.48%) 


In [7]:
# Conteo de Duplicados
duplicados = df.duplicated().sum()
print(f"Filas totalmente duplicadas: {duplicados}")

# Duplicados por Client ID
duplicados_id = df['client_id'].duplicated().sum()
print(f"Clientes repetidos: {duplicados_id}")

Filas totalmente duplicadas: 0
Clientes repetidos: 0


- Nulos Críticos: Identificados ***20,109*** usuarios 
- Desbalanceo de Grupos:
   Test: ***26,968*** usuarios.
   Control: ***23,532*** usuarios.
- Renombrado de columna Variation a experiment_group.

In [9]:
df.to_csv("df_experiment_final.csv", index=False)

